### PR TITLE
Improve `build-lint-test` workflow

### PR DESCRIPTION
**Pull request by gantunesr**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -62,7 +62,7 @@
     "eslint-plugin-node": "^11.1.0",
     "eslint-plugin-prettier": "^4.2.1",
     "ethereumjs-wallet": "^1.0.1",
-    "jest": "^29.0.0",
+    "jest": "^29.5.0",
```
Comment (Gudahtt) [inline package.json:65]: Is this related to the workflow improvements?

**Review by Gudahtt — APPROVED** [review]
LGTM!

File: package.json (hunk 1 above)
Comment (gantunesr) [reply package.json:65]: No, I was testing to see if it could be related to the problem. I can revert it since it's not relevant to the purpose of this PR

Comment (gantunesr) [reply package.json]: Done in [4ffc57d](https://github.com/MetaMask/KeyringController/pull/221/commits/4ffc57d0a78f8b93f2ef193c39d2564bc796ded8)

**Review by Gudahtt — APPROVED** [review]
